search/search.go

```diff
@@ -265,12 +265,14 @@ func (dm *DocumentMatch) Complete(prealloc []Location) []Location {
 		}
 		prealloc = prealloc[:nlocs]
 
+		var firstField = true
 		var lastField string
 		var tlm TermLocationMap
 		var needsDedupe bool
 
 		for i, ftl := range dm.FieldTermLocations {
-			if lastField != ftl.Field {
```

Comment:
Hmm, wouldn't have guessed that `lastField != ftl.Field` was simply only a zero value check. Would've guessed that it could do with a redundant field check as well (if array was sorted).

How about just checking for `i == 0` instead of introducing the new variable `firstField` for this?

Reply:
Oh, sorry if I gave that impression.  It is **NOT** simply/only a zero value check.  It is intended to (and functions correctly) process a list sorted by field.  Thus every time a new field is encountered, some additional work must be done.  The check correctly detects that.

However, the author of this code believed that the empty field name was not valid, therefore we will never encounter an empty field name, except on that first run.  The code actually relies on this for correctness, because the initializaiton of `tlm` only happens when cross this "i see a new field" boundary.  So now, if we think the authors assumption was not valid, and we want to allow for empty string, we cannot rely on the zero value to track the state for us.

Reply:
And yes, we can use `i`, good idea.